packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/LightCurveViewer/LightCurveViewerContainer.js

```diff
@@ -28,9 +28,9 @@ class LightCurveViewerContainer extends Component {
   }
 
   componentDidUpdate (prevProps) {
-    const prevSubject = prevProps.subject
-    const { subject } = this.props
-
+    const prevSubject = prevProps.subject.toJSON()
+    const subject = this.props.subject.toJSON()
```

Comment:
The MST `toJSON()` method is meant for debugging according to their docs. I've been using the MobX `toJS()` for conversion instead which also gives us a configurable option about what to do with maps: https://mobx.js.org/refguide/tojson.html

I might suggest adding a getter to the store's view methods that returns the vanilla JS as needed.

Reply:
Unfortunately, that isn't working here, since the object referred to by `prevProps` has already been removed from the store. What I did instead was pass `subject.id` as an additional prop to the current `Viewer` component, which works as it's a primitive.

Reply:
Ah right, that makes sense.